src/WebJobs.Script/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -70,20 +70,8 @@ public static TValue GetRequestPropertyOrDefault<TValue>(this HttpRequestMessage
 
         public static IDictionary<string, string> GetRawHeaders(this HttpRequestMessage request)
         {
-            Dictionary<string, string> headers = new Dictionary<string, string>();
-
-            var allHeadersRaw = request.Headers.ToString();
-            var rawHeaderLines = allHeadersRaw.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
-
-            foreach (var header in rawHeaderLines)
-            {
-                int idx = header.IndexOf(':');
-                string name = header.Substring(0, idx);
-                string value = header.Substring(idx + 1).Trim();
-                headers.Add(name, value);
-            }
-
-            return headers;
+            var headers = request.Content?.Headers.Concat(request.Headers) ?? request.Headers;
+            return headers.ToDictionary(p => p.Key, p => string.Join(" ", p.Value));
```

Comment:
Great question, after looking at [this httpheaders comment](https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Net.Http/src/System/Net/Http/Headers/ProductInfoHeaderParser.cs#L13) I'm going to go back to the original serializing code.  Not worth it trying to simplify if there are weird edge cases like this.